app/controllers/api/v2/discovered_hosts_controller.rb

```diff
@@ -76,19 +76,12 @@ def create
       end
 
       def update
-        @host                           = @discovered_host.becomes(::Host::Managed)
-        @host.type                      = 'Host::Managed'
-        @host.managed                   = true
-        @host.primary_interface.managed = true
-        @host.build                     = true
-        forward_request_url
-        update_response = @host.update_attributes(params[:discovered_host])
         Host.transaction do
-          if update_response
-            delete_discovery_attribute_set(@host.id)
-          end
+          @host = ::ForemanDiscovery::HostConverter.to_managed(@discovered_host)
+          forward_request_url
+          update_response = @host.update_attributes(params[:discovered_host])
+          process_response update_response
         end
-        process_response update_response
       end
```

Comment:
@lzap it fixes the token bug at least for my scenario.
I'm closing the other PR